src/lib/components/common/Interest.svelte

```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+    import GradientCard from './GradientCard.svelte';
+
+    interface Interest {
+        id: number;
+        name: string;
+        icon: string;
+        selected: boolean;
+        bgFrom: string;
+        bgTo: string;
+    }
+
+    let interests: Interest[] = [
+        {
+            id: 1,
+            name: 'Sport & Travel',
+            icon: 'icon-[system-uicons--button-minus]',
+            selected: false,
+            bgFrom: 'from-red-200',
+            bgTo: 'to-red-700'
+        },
+        {
+            id: 2,
+            name: 'Books',
+            icon: 'icon-[system-uicons--button-minus]',
+            selected: false,
+            bgFrom: 'from-pink-200',
+            bgTo: 'to-pink-800'
+        },
+        {
+            id: 3,
+            name: 'Men fashion',
+            icon: 'icon-[system-uicons--button-minus]',
+            selected: false,
+            bgFrom: 'from-yellow-200',
+            bgTo: 'to-pink-900'
+        },
+        {
+            id: 4,
+            name: 'Women handbag',
+            icon: 'icon-[system-uicons--button-minus]',
+            selected: false,
+            bgFrom: 'from-yellow-200',
+            bgTo: 'to-green-700'
+        },
+        {
+            id: 5,
+            name: 'Women fashion',
+            icon: 'icon-[system-uicons--button-minus]',
+            selected: false,
+            bgFrom: 'from-yellow-200',
+            bgTo: 'to-pink-900'
+        },
+        {
+            id: 6,
+            name: 'Men handbag',
+            icon: 'icon-[system-uicons--button-minus]',
+            selected: false,
+            bgFrom: 'from-yellow-200',
+            bgTo: 'to-green-700'
+        }
+    ];
+
+    export let displayMode: string = 'grid'; // Default display mode is grid
+
+    function toggleSelection(id: number) {
+        interests = interests.map((item) =>
+            item.id === id ? { ...item, selected: !item.selected } : item
+        );
+    }
+</script>
+
+<div class="">
+    <div class="{displayMode === 'grid' ? 'grid grid-cols-2 gap-3 w-full pl-3 pr-3 mr-10 h-97 overflow-y-auto' : 'flex flex-col w-full pl-3 pr-3 gap-3 h-97 overflow-y-auto'}">
```

Comment:
some classes are similar between this condition.
This makes code reduntdant and longer